scrapy/pqueues.py

```diff
@@ -105,12 +109,6 @@ def pop(self):
         return m
 
     def peek(self):
-        """Returns the next object to be returned by :meth:`pop`,
-        but without removing it from the queue.
-
-        Raises :exc:`NotImplementedError` if the underlying queue class does
-        not implement a ``peek`` method, which is optional for queues.
-        """
```

Comment:
Accidental removal of documentation?

Reply:
It isn't accidental. Corresponding documentation moved to `docs/topics/request-queues.rst`

Reply:
With the recent changes about `NotImplementedError` and `AttributeError`, I’ve reviewed this again and I think this docstring needs to stay.

This is not the documentation of a disk downstream queue, which is the context where `docs/topics/request-queues.rst` documents the optional `peek` method. This is a priority queue, and at the moment we don’t seem to be documenting the interface of these like we do for downstream disk queues, we only document the 2 specific implementations of priority queues that Scrapy provides, so we should keep this docstring, both here and in `DownloaderAwarePriorityQueue.peek`.

In the future we may document the interface of priority queues and memory downstream queues, which I assume is out of the scope of these changes. When we do that, we can remove these docstrings, and replace them by a note at the class-level docstring that indicates whether or not peek is supported.